core/src/docs/rfcs/5314_remove_metakey.md

```diff
@@ -0,0 +1,120 @@
+- Proposal Name: `remove_metakey`
+- Start Date: 2024-11-12
+- RFC PR: [apache/opendal#5313](https://github.com/apache/opendal/pull/5313)
+- Tracking Issue: [apache/opendal#5314](https://github.com/apache/opendal/issues/5314)
+
+# Summary
+
+Remove the `Metakey` concept from OpenDAL and replace it with a simpler and more predictable metadata handling mechanism.
+
+# Motivation
+
+The current `Metakey` design has several issues:
+
+1. Performance Impact: Users often initiate costly operations unintentionally, such as using `Metakey::Full`, which results in extra stat calls
+2. Usability Issues: Users often try to access metadata that hasn't been explicitly requested
+3. API Confusion: There's a conflict between `Metakey::Version` and the new `version(bool)` parameter
+4. Implementation Complexity: Service developers struggle to implement `Metakey` correctly
+
+The goal is a simpler, more intuitive API that prevents common mistakes and improves performance as standard.
+
+# Guide-level explanation
+
+Instead of using `Metakey` to specify which metadata fields to fetch, services will now declare their metadata capabilities upfront through a new `MetadataCapability` struct:
+
+```rust
+let entries = op.list("path").await?;
+for entry in entries {
+    if op.metadata_capability().content_type {
+        println!("Content-Type: {}", entry.metadata().content_type());
+    }
+}
+```
+
+If users need additional metadata not provided by `list`:
+
+```rust
+let entries = op.list("path").await?;
+for entry in entries {
+    let mut meta = entry.metadata();
+    if !op.metadata_capability().etag {
```

Comment:
Based on this usage, it seems to be the metadata capability of the `list` operation ?